### ISSUE TITLE
"get snapshot failed" after "create ccr success"

### ISSUE THREAD

**Issue by Jw9394** [selectdb/ccr-syncer#26]
source cluster version: doris-2.0.1-rc04
destination cluster version: doris-2.0.1-rc04
ccr-syncer commit id 69d5f24

i am trying to copy a table from cluster a to cluster b, so i want to use ccr-syncer to achieve that.

the curl request body is like:
```
curl -X POST -H "Content-Type: application/json" -d '{
    "name": "ccr_test",
    "src": {
    "host": "192.168.1.10",
    "port": "9030",
    "thrift_port": "9020",
    "user": "root",
    "password": "123456",
    "database": "testdb1",
    "table": "testtb1"
    },
    "dest": {
    "host": "192.168.2.10",
    "port": "9030",
    "thrift_port": "9020",
    "user": "root",
    "password": "123456",
    "database": "testdb2",
    "table": "testtb2"
    }
}' http://192.168.2.10:9190/create_ccr
```
after that, ccr return "create ccr success". this ccr process will not success because this error:
```
get snapshot failed, status: TStatus({StatusCode:SNAPSHOT_NOT_EXIST ErrorMsgs:[snapshot not exist]}) job=ccr_test line=ccr/job.go:319
[2023-09-20 07:53:18] DEBUG job:  job=ccr_test line=ccr/job.go:322
```
the full ccr log is:
```
start time: Wed Sep 20 07:52:49 AM UTC 2023
[2023-09-20 07:52:49]  INFO Server listening on :9190 line=service/http_service.go:275
[2023-09-20 07:52:50] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:52:50] DEBUG checker state: checkerStateUpdate line=ccr/checker.go:135
[2023-09-20 07:52:50]  INFO update jobs [] line=ccr/checker.go:119
[2023-09-20 07:52:50] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:52:50] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:52:55] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:52:55] DEBUG checker state: checkerStateCheck line=ccr/checker.go:135
[2023-09-20 07:52:55] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:52:57]  INFO create ccr line=service/http_service.go:83
[2023-09-20 07:52:57]  INFO create ccr name: ccr_test, src: {{192.168.1.10 9030 9020} [] root 123456  testdb1 0 testtb1 0 []}, dest: {{192.168.2.10 9030 9020} [] root 123456  testdb2 0 testtb2 0 []} line=service/http_service.go:49
[2023-09-20 07:52:57]  INFO add job: ccr_test line=ccr/job_manager.go:43
[2023-09-20 07:52:57]  INFO first run check job{src 15 0 host: 192.168.1.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb1, database id: 0, table: testtb1, table id: 0 <nil>} {dest 15 0 host: 192.168.2.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb2, database id: 0, table: testtb2, table id: 0 <nil>} line=ccr/job.go:1298
[2023-09-20 07:52:57] DEBUG show proc '/backends' line=ccr/meta.go:449
[2023-09-20 07:52:57] DEBUG backend: Backend: {Id: 10002, Host: 192.168.1.11, HeartbeatPort: 9050, BePort: 9060, HttpPort: 8040, BrpcPort: 8060} line=ccr/meta.go:494
[2023-09-20 07:52:57] DEBUG backend: Backend: {Id: 10003, Host: 192.168.1.12, HeartbeatPort: 9050, BePort: 9060, HttpPort: 8040, BrpcPort: 8060} line=ccr/meta.go:494
[2023-09-20 07:52:57] DEBUG backend: Backend: {Id: 10004, Host: 192.168.1.13, HeartbeatPort: 9050, BePort: 9060, HttpPort: 8040, BrpcPort: 8060} line=ccr/meta.go:494
[2023-09-20 07:52:57] DEBUG show proc '/backends' line=ccr/meta.go:449
[2023-09-20 07:52:57] DEBUG backend: Backend: {Id: 10019, Host: 127.0.0.1, HeartbeatPort: 9050, BePort: 9060, HttpPort: 8040, BrpcPort: 8060} line=ccr/meta.go:494
[2023-09-20 07:52:57] DEBUG check database exist by spec{spec 15 0 host: 192.168.1.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb1, database id: 0, table: testtb1, table id: 0 <nil>} line=base/spec.go:383
[2023-09-20 07:52:57] DEBUG check table exists by spec{spec 15 0 host: 192.168.1.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb1, database id: 0, table: testtb1, table id: 0 <nil>} line=base/spec.go:417
[2023-09-20 07:52:57]  INFO check table testdb1.testtb1 enable binlog line=base/spec.go:231
[2023-09-20 07:52:57]  INFO table testdb1.testtb1 create string: CREATE TABLE `testtb1` (
  `id` int(11) NOT NULL,
  `col1` varchar(33) NOT NULL,
  `col2` varchar(60) NULL,
  `col3` varchar(60) NULL
) ENGINE=OLAP
UNIQUE KEY(`id`)
COMMENT 'test'
DISTRIBUTED BY HASH(`id`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"is_being_synced" = "false",
"storage_format" = "V2",
"light_schema_change" = "true",
"disable_auto_compaction" = "false",
"binlog.enable" = "true",
"binlog.ttl_seconds" = "864000",
"binlog.max_bytes" = "9223372036854775807",
"binlog.max_history_nums" = "9223372036854775807",
"enable_single_replica_compaction" = "false"
); line=base/spec.go:261
[2023-09-20 07:52:57]  INFO UpdateTable tableName: testtb1, tableId: 0 line=ccr/meta.go:107
[2023-09-20 07:52:57] DEBUG found table:testdb1.testtb1, tableId:44079 line=ccr/meta.go:152
[2023-09-20 07:52:57] DEBUG check database exist by spec{spec 15 0 host: 192.168.2.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb2, database id: 0, table: testtb2, table id: 0 <nil>} line=base/spec.go:383
[2023-09-20 07:52:57] DEBUG check table exists by spec{spec 15 0 host: 192.168.2.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb2, database id: 0, table: testtb2, table id: 0 <nil>} line=base/spec.go:417
[2023-09-20 07:52:57]  INFO new snapshot, commitSeq: 0 job=ccr_test line=ccr/job.go:1235
[2023-09-20 07:52:57] DEBUG update job progress: JobProgress{JobName: ccr_test, SyncState: TableFullSync, SubSyncState: BeginCreateSnapshot, CommitSeq: 0, TableCommitSeqMap: map[], InMemoryData: <nil>, PersistData: } job=ccr_test line=ccr/job_progress.go:285
[2023-09-20 07:52:57] DEBUG update job progress done: JobProgress{JobName: ccr_test, SyncState: TableFullSync, SubSyncState: BeginCreateSnapshot, CommitSeq: 0, TableCommitSeqMap: map[], InMemoryData: <nil>, PersistData: } job=ccr_test line=ccr/job_progress.go:308
[2023-09-20 07:53:00] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:53:00] DEBUG checker state: checkerStateCheck line=ccr/checker.go:135
[2023-09-20 07:53:00] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:53:00] DEBUG table full sync job=ccr_test line=ccr/job.go:1158
[2023-09-20 07:53:00]  INFO fullsync status: create snapshot job=ccr_test line=ccr/job.go:277
[2023-09-20 07:53:00]  INFO create snapshot testdb1.ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:472
[2023-09-20 07:53:00] DEBUG backup snapshot sql: BACKUP SNAPSHOT testdb1.ccrs_testdb1_testtb1_1695196380 TO `__keep_on_local__` ON ( testtb1 ) PROPERTIES ("type" = "full") job=ccr_test line=base/spec.go:480
[2023-09-20 07:53:00] DEBUG check backup state{database 15 0 testdb1 <nil>} job=ccr_test line=base/spec.go:533
[2023-09-20 07:53:00] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:00] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:00]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [PENDING] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:03] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:03] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:03]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [SNAPSHOTING] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:05] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:53:05] DEBUG checker state: checkerStateCheck line=ccr/checker.go:135
[2023-09-20 07:53:05] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:53:06] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:06] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:06]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [UPLOAD_SNAPSHOT] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:09] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:09] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:09]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [UPLOADING] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:10] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:53:10] DEBUG checker state: checkerStateCheck line=ccr/checker.go:135
[2023-09-20 07:53:10] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:53:12] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:12] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:12]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [SAVE_META] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:15] DEBUG checker state: checkerStateRefresh line=ccr/checker.go:135
[2023-09-20 07:53:15] DEBUG checker state: checkerStateCheck line=ccr/checker.go:135
[2023-09-20 07:53:15] DEBUG checker state: checkerStateFinish line=ccr/checker.go:135
[2023-09-20 07:53:15] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:15] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:15]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [UPLOAD_INFO] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:18] DEBUG check backup state of snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=base/spec.go:500
[2023-09-20 07:53:18] DEBUG check backup state sql: SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380" job=ccr_test line=base/spec.go:508
[2023-09-20 07:53:18]  INFO check snapshot ccrs_testdb1_testtb1_1695196380 backup state: [FINISHED] job=ccr_test line=base/spec.go:526
[2023-09-20 07:53:18] DEBUG update job progress: JobProgress{JobName: ccr_test, SyncState: TableFullSync, SubSyncState: GetSnapshotInfo, CommitSeq: 0, TableCommitSeqMap: map[], InMemoryData: <nil>, PersistData: ccrs_testdb1_testtb1_1695196380} job=ccr_test line=ccr/job_progress.go:285
[2023-09-20 07:53:18] DEBUG update job progress done: JobProgress{JobName: ccr_test, SyncState: TableFullSync, SubSyncState: GetSnapshotInfo, CommitSeq: 0, TableCommitSeqMap: map[], InMemoryData: <nil>, PersistData: ccrs_testdb1_testtb1_1695196380} job=ccr_test line=ccr/job_progress.go:308
[2023-09-20 07:53:18]  INFO fullsync status: get snapshot info job=ccr_test line=ccr/job.go:303
[2023-09-20 07:53:18] DEBUG begin get snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=ccr/job.go:312
[2023-09-20 07:53:18] DEBUG GetSnapshot ccrs_testdb1_testtb1_1695196380, spec: host: 192.168.1.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb1, database id: 44077, table: testtb1, table id: 44079 job=ccr_test line=rpc/fe.go:213
[2023-09-20 07:53:18] DEBUG GetSnapshotRequest user root, db testdb1, table testtb1, label name ccrs_testdb1_testtb1_1695196380, snapshot name , snapshot type 1 job=ccr_test line=rpc/fe.go:226
[2023-09-20 07:53:18] ERROR get snapshot failed, status: TStatus({StatusCode:SNAPSHOT_NOT_EXIST ErrorMsgs:[snapshot not exist]}) job=ccr_test line=ccr/job.go:319
[2023-09-20 07:53:18] DEBUG job:  job=ccr_test line=ccr/job.go:322
[2023-09-20 07:53:18] ERROR job sync failed, job: ccr_test, err: jobInfo is not set
github.com/selectdb/ccr_syncer/ccr.(*Job).fullSync
        /root/ccr-syncer/ccr/job.go:324
github.com/selectdb/ccr_syncer/ccr.(*Job).fullSync
        /root/ccr-syncer/ccr/job.go:473
github.com/selectdb/ccr_syncer/ccr.(*Job).tableSync
        /root/ccr-syncer/ccr/job.go:1159
github.com/selectdb/ccr_syncer/ccr.(*Job).sync
        /root/ccr-syncer/ccr/job.go:1208
github.com/selectdb/ccr_syncer/ccr.(*Job).run
        /root/ccr-syncer/ccr/job.go:1227
github.com/selectdb/ccr_syncer/ccr.(*Job).Run
        /root/ccr-syncer/ccr/job.go:1289
github.com/selectdb/ccr_syncer/ccr.(*JobManager).runJob.func1
        /root/ccr-syncer/ccr/job_manager.go:155
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1598 job=ccr_test line=ccr/job.go:1228
[2023-09-20 07:53:18] DEBUG table full sync job=ccr_test line=ccr/job.go:1158
[2023-09-20 07:53:18]  INFO fullsync status: get snapshot info job=ccr_test line=ccr/job.go:303
[2023-09-20 07:53:18] DEBUG begin get snapshot ccrs_testdb1_testtb1_1695196380 job=ccr_test line=ccr/job.go:312
[2023-09-20 07:53:18] DEBUG GetSnapshot ccrs_testdb1_testtb1_1695196380, spec: host: 192.168.1.10, port: 9030, thrift_port: 9020, user: root, cluster: , database: testdb1, database id: 44077, table: testtb1, table id: 44079 job=ccr_test line=rpc/fe.go:213
[2023-09-20 07:53:18] DEBUG GetSnapshotRequest user root, db testdb1, table testtb1, label name ccrs_testdb1_testtb1_1695196380, snapshot name , snapshot type 1 job=ccr_test line=rpc/fe.go:226
[2023-09-20 07:53:18] ERROR get snapshot failed, status: TStatus({StatusCode:SNAPSHOT_NOT_EXIST ErrorMsgs:[snapshot not exist]}) job=ccr_test line=ccr/job.go:319
[2023-09-20 07:53:18] DEBUG job:  job=ccr_test line=ccr/job.go:322
[2023-09-20 07:53:18] ERROR job sync failed, job: ccr_test, err: jobInfo is not set
github.com/selectdb/ccr_syncer/ccr.(*Job).fullSync
        /root/ccr-syncer/ccr/job.go:324
github.com/selectdb/ccr_syncer/ccr.(*Job).tableSync
        /root/ccr-syncer/ccr/job.go:1159
github.com/selectdb/ccr_syncer/ccr.(*Job).sync
        /root/ccr-syncer/ccr/job.go:1208
github.com/selectdb/ccr_syncer/ccr.(*Job).run
        /root/ccr-syncer/ccr/job.go:1227
github.com/selectdb/ccr_syncer/ccr.(*Job).Run
        /root/ccr-syncer/ccr/job.go:1289
github.com/selectdb/ccr_syncer/ccr.(*JobManager).runJob.func1
        /root/ccr-syncer/ccr/job_manager.go:155
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1598 job=ccr_test line=ccr/job.go:1228
```

**Comment by JackDrogon** [comment]
can u show me more about fe log. which is the network topology about your doris and syncer?
u can also connect doris fe to `SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380"`

**Comment by Jw9394** [comment]
> can u show me more about fe log. which is the network topology about your doris and syncer? u can also connect doris fe to `SHOW BACKUP FROM testdb1 WHERE SnapshotName = "ccrs_testdb1_testtb1_1695196380"`

the log from source fe/be or the destination fe/be ?

**Comment by Jw9394** [comment]
apache-doris released 2.0.2 version yesterday and i upgrade my cluster from 2.0.1 to 2.0.2 today. now ccr can print the actual error.
```
[2023-10-08 02:29:40] ERROR get snapshot failed, status: TStatus({StatusCode:NOT_MASTER ErrorMsgs:[FE is not master]}) job=ccr_test
[2023-10-08 02:29:40] ERROR job sync failed, job: ccr_test, err: jobInfo is not set
```
i change my curl request destination to fe master node and it works fine.
